tests/console/docker.pm

```diff
@@ -50,47 +50,62 @@ sub run {
     validate_script_output("docker search --no-trunc opensuse", sub { m/This project contains the stable releases of the openSUSE distribution/ });
 
     # images can be pulled from the Docker Hub
-    #   - pull minimalistic alpine image of declared version
+    #   - pull minimalistic alpine image of declared version using tag
     #   - https://store.docker.com/images/alpine
     my $alpine_image_version = '3.5';
     assert_script_run("docker pull alpine:$alpine_image_version", 300);
-    assert_script_run("docker images -q alpine:$alpine_image_version");
-    #   - pull hello-world image, typical docker demo image
+    assert_script_run("docker images alpine:$alpine_image_version | grep alpine");
+    #   - pull typical docker demo image without tag. Should be latest.
     #   - https://store.docker.com/images/hello-world
     assert_script_run("docker pull hello-world", 300);
-    assert_script_run("docker images -q hello-world:latest");
+    assert_script_run("docker images hello-world | grep latest");
 
     # local images can be listed
-    validate_script_output("docker images", sub { m/alpine/ });
+    my $cmd_docker_images = 'docker images';
+    my $listed_images     = script_output($cmd_docker_images);
+    unless ($listed_images =~ m/alpine/ && $listed_images =~ m/hello-world/) {
+        die("error: missing image: $cmd_docker_images");
```

Comment:
you will die with error ""error: missing image: docker images"
while you probably want to die with something like ""error: missing image: alpine or hello-world"

```
 my $listed_images     = script_output('docker images');
die("missing image: alpine") if $listed_images =~ m/alpine/; 
die("missing image: hello-world") if $listed_images =~ m/hello-world/; 

```
 what about  this version ?

Reply:
@drpaneas, here I am not testing if pull worked, but if `docker images` works. Extending coverage.

Reply:
Earlier in the code, you are doing this:

```
assert_script_run("docker images hello-world | grep latest");
assert_script_run("docker images alpine:$alpine_image_version | grep alpine");
```

how this is different? I mean if `docker images` doesn't work, the test should have failed earlier. No?